### DISCUSSION TITLE
delete a device service

### DISCUSSION THREAD

**Discussion by collateral87** [frenck/spook#133] [Ideas]
Hey frenck,
How about a service to delete devices?
That would be super cool 😎

Thanks for this nice and funny addon

**Comment by frenck** [comment]
Oh hi @collateral87 👋 

> How about a service to delete devices?

Hmm... What would that do? As in, deleting an device is only possible is very limited circumstances.
Or are you asking for a service that provides the ability to delete a device, for which also the **"Delete device"** button in the UI is shown?

> Thanks for this nice and funny addon

You're welcome! ❤️ 
_PS: This is not an add-on, but an integration 😉_

../Frenck

**Comment by collateral87** [comment]
There are some integrations whose devices cannot be deleted and do not delete themselves. SmartThings is an example of this. However, I did not know that special requirements apply for this.
That sounds like bad news then or is there a chance? 🧐

> PS: This is not an add-on, but an integration 😉

Please show mercy 🙏

**Reply by frenck** [reply]
> Please show mercy 🙏

<img src="https://media4.giphy.com/media/LPdTYfGETFqyQ/giphy.gif"/>

🤣

**Reply by frenck** [reply]
> There are some integrations whose devices cannot be deleted and do not delete themselves. SmartThings is an example of this. However, I did not know that special requirements apply for this.
That sounds like bad news then or is there a chance? 🧐

So, integration should/can provide the ability if that applies to them. But, ideally, integration should just clean up after their own 😄 

I guess, I could start by implementing a service that mimics the existing **Delete device** button (including its rules), and add a possible `force` flag to allow forcibly deletion if you'd really like to delete regardless anything.

**Reply by collateral87** [reply]
> So, integration should/can provide the ability if that applies to them. But, ideally, integration should just clean up after their own 😄

I completely agree with you, but unfortunately, that's often not the case, so...
...yes, yes, and yes please!

That would be amazing 🤩 (and scary 👻)
